src/NHibernate.Test/TestCase.cs

```diff
@@ -313,6 +315,15 @@ protected virtual DebugSessionFactory BuildSessionFactory()
 		private void Cleanup()
 		{
 			Sfi?.Close();
+
+			// Clear connection pool for Oracle to avoid problem that was manifested with https://github.com/nhibernate/nhibernate-core/pull/1517:
+			// As it seems Oracle can cache returned types for query for given connection.
+			// So exception can be thrown if two tests execute same query but with different types in result (like for Entity.Id int and Entity.Id Guid)
+			if (Dialect is Oracle8iDialect)
+			{
+				OracleConnection.ClearAllPools();
```

Comment:
This will work only if the Oracle driver is the managed one. But two other Oracle drivers are usable. (Currently their specific TeamCity builds are not enabled.)

I have made a PR on your branch for handling them all: bahusoid/nhibernate-core#1.

Reply:
Thanks. I've adjusted it per @hazzik comments